graphql/directive_test.go

```diff
@@ -216,7 +232,118 @@ func setupPermissions(t *testing.T) {
+			obj := any(map[string]any{"hostId": "host1"})
+			_, err := config.Directives.RequireHostAccess(ctx, obj, next, HostAccessLevelEdit)
+			assert.EqualError(t, err, "input: user 'test_user' does not have permission to access host 'host1'")
+		},
+		"ViewSucceedsWhenUserHasViewPermission": func(ctx context.Context, t *testing.T, next func(rctx context.Context) (any, error), config Config, usr *user.DBUser) {
+			assert.NoError(t, usr.AddRole(ctx, "view_host-id"))
+			nextCalled := false
+			wrappedNext := func(rctx context.Context) (any, error) {
+				nextCalled = true
+				return nil, nil
+			}
+			obj := any(map[string]any{"hostId": "host1"})
+			res, err := config.Directives.RequireHostAccess(ctx, obj, wrappedNext, HostAccessLevelView)
+			assert.NoError(t, err)
+			assert.Nil(t, res)
+			assert.Equal(t, true, nextCalled)
+			assert.NoError(t, usr.RemoveRole(ctx, "view_host-id"))
+		},
+		"EditSucceedsWhenUserHasEditPermission": func(ctx context.Context, t *testing.T, next func(rctx context.Context) (any, error), config Config, usr *user.DBUser) {
+			assert.NoError(t, usr.AddRole(ctx, "edit_host-id"))
+			nextCalled := false
+			wrappedNext := func(rctx context.Context) (any, error) {
+				nextCalled = true
+				return nil, nil
+			}
+			obj := any(map[string]any{"hostId": "host1"})
+			res, err := config.Directives.RequireHostAccess(ctx, obj, wrappedNext, HostAccessLevelEdit)
+			assert.NoError(t, err)
+			assert.Nil(t, res)
+			assert.Equal(t, true, nextCalled)
+			assert.NoError(t, usr.RemoveRole(ctx, "edit_host-id"))
+		},
+		"ViewSucceedsWhenHostIsStartedByUser": func(ctx context.Context, t *testing.T, next func(rctx context.Context) (any, error), config Config, usr *user.DBUser) {
+			nextCalled := false
+			wrappedNext := func(rctx context.Context) (any, error) {
+				nextCalled = true
+				return nil, nil
+			}
+			obj := any(map[string]any{"hostId": "host2"})
+			res, err := config.Directives.RequireHostAccess(ctx, obj, wrappedNext, HostAccessLevelView)
+			assert.NoError(t, err)
+			assert.Nil(t, res)
+			assert.Equal(t, true, nextCalled)
+		},
+		"EditSucceedsWhenHostIsStartedByUser": func(ctx context.Context, t *testing.T, next func(rctx context.Context) (any, error), config Config, usr *user.DBUser) {
+			nextCalled := false
+			wrappedNext := func(rctx context.Context) (any, error) {
+				nextCalled = true
+				return nil, nil
+			}
+			obj := any(map[string]any{"hostId": "host2"})
+			res, err := config.Directives.RequireHostAccess(ctx, obj, wrappedNext, HostAccessLevelEdit)
+			assert.NoError(t, err)
+			assert.Nil(t, res)
+			assert.Equal(t, true, nextCalled)
+		},
+	} {
+		t.Run(tName, func(t *testing.T) {
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+			setupPermissions(t)
+			usr, err := setupUser(t)
+			assert.NoError(t, err)
+			assert.NotNil(t, usr)
+			ctx = gimlet.AttachUser(ctx, usr)
```

Comment:
From what I can tell it looks like we give the user all the permissions and then run these tests. Can we also add tests that test if the user is missing the necessary permissions?

Reply:
Those tests exist [here](https://github.com/evergreen-ci/evergreen/pull/8784/files#diff-9c5fad79d640767488dc9a197b59698f179b9b17df2c8bbfa04c6b3172cd61c9R251-R256)

Reply:
Thanks I missed the difference in ids tied to the scope